### PR TITLE
add DfmCodeRenderer.

### PR DESCRIPTION
**Pull request by vwxyzh**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Dfm/DfmRenderer.cs [hunk 1]
```diff
@@ -153,30 +153,7 @@ public override StringBuffer Render(IMarkdownRenderer renderer, MarkdownBlockquo
...
-            }
-            catch (DirectoryNotFoundException)
-            {
-                return DfmFencesBlockHelper.GenerateReferenceNotFoundErrorMessage(renderer, token);
-            }
-            catch (FileNotFoundException)
-            {
-                return DfmFencesBlockHelper.GenerateReferenceNotFoundErrorMessage(renderer, token);
-            }
+            return _codeRenderer.Render(renderer, token, context);
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Dfm/DfmRenderer.cs]: >_codeRenderer [](start = 19, length = 13)

need opdfm renderer changes as well?

File: src/Microsoft.DocAsCode.Dfm/DfmCodeRenderer.cs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                }
+                result += "</code></pre>";
+            }
+
+            return result;
+        }
+
+        public virtual StringBuffer RenderPreTag(StringBuffer result, DfmFencesToken token, Options options)
+        {
+            result += "<pre>";
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Dfm/DfmCodeRenderer.cs]: >pre [](start = 24, length = 3)

need we also provide render pre end tag?

File: src/Microsoft.DocAsCode.Dfm/DfmCodeRenderer.cs [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+            if (codeLines != null)
+            {
+                result = RenderPreTag(result, token, options);
+                result = RenderCodeTag(result, token, options);
+                foreach (var line in codeLines)
+                {
+                    result += StringHelper.HtmlEncode(line);
+                    result += "\n";
+                }
+                result += "</code></pre>";
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Dfm/DfmCodeRenderer.cs]: ></code></pre> [](start = 27, length = 13)

need these be extracted?

File: src/Microsoft.DocAsCode.Dfm/DfmRenderer.cs (hunk 1 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Dfm/DfmRenderer.cs]: Yes, I will do it tomorrow.

**Comment by vicancy** [comment]
:shipit:

**Pull request merged**